src/rp2_common/hardware_clocks/include/hardware/clocks.h

```diff
@@ -172,22 +172,22 @@ void clocks_enable_resus(resus_callback_t resus_callback);
  *
  * \param gpio The GPIO pin to output the clock to. Valid GPIOs are: 21, 23, 24, 25. These GPIOs are connected to the GPOUT0-3 clock generators.
  * \param src  The source clock. See the register field CLOCKS_CLK_GPOUT0_CTRL_AUXSRC for a full list. The list is the same for each GPOUT clock generator.
- * \param div  The amount to divide the source clock by. This is useful to not overwhelm the GPIO pin with a fast clock.
- * \param frac The frac to divide the source clock by.
+ * \param div_int  The integer part of dividing the source clock by. This is useful to not overwhelm the GPIO pin with a fast clock. this is in range of 1..2^24-1.
```

Comment:
Could add an `invalid_params_if` to the .c file to check that `div_int < (1<<24)` ? With the 24 calculated as `CLOCKS_CLK_GPOUT0_DIV_INT_MSB - CLOCKS_CLK_GPOUT0_DIV_INT_LSB + 1` ?
(but if @kilograham disagrees, please listen to him rather than me :laughing: )